### PR TITLE
Reintegrate skeleton-docker into repository

### PR DESCRIPTION
**Pull request by zenine07**
# 

## 🗣 Description

This PR would reintegrate the [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker) skeleton into this repository.

This was done by adding the correct skeleton repository as a remote, pulling from its develop branch, and resolving any conflicts shown. 

```console
git remote add skeleton https://github.com/cisagov/skeleton-docker.git
git remote set-url --push skeleton no_push
git pull --allow-unrelated-histories skeleton develop
git status
```

After skeleton was merged, fixed pre-commit issues that showed up.

## 💭 Motivation and Context

When this project was created, it was not started using the [`skeleton` tool](https://github.com/cisagov/development-guide/tree/develop/project_setup#skeleton-tool-) This prevents the [`lineage` GitHub Action](https://github.com/cisagov/action-lineage) from automatically generating PRs to ensure downstream repositories are kept up-to-date.

## 🧪 Testing

Automated tests pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)
- [x] Merge with skeleton

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by zenine07** [comment]
@dav3r Can you give this repo access to the organization secrets, so the last check can pass?

**Comment by dav3r** [comment]
> @dav3r Can you give this repo access to the organization secrets, so the last check can pass?

Done and done!

File: src/utils/proxies/websense.py
Comment (jsf9k) [inline src/utils/proxies/websense.py]:
```suggestion
    # Bandit complains that url is not being audited as it will open file:/ or custom schemes.
```

File: src/secrets/quote.txt
Comment (jsf9k) [inline src/secrets/quote.txt]: Is this file being used?  If not it should be removed.

File: requirements.txt
Comment (jsf9k) [inline requirements.txt]: Is this dependency needed here, or could it be moved to `requirements-test.txt`?

File: requirements-test.txt
Comment (jsf9k) [inline requirements-test.txt:3]: This seems like a test thing that should be here instead of in `requirements.txt`.

File: CONTRIBUTING.md
Comment (jsf9k) [inline CONTRIBUTING.md]: Change _all_ references to "cisagov/skeleton-docker" in this file to "cisagov/domain-manager-api".

**Review by jsf9k — CHANGES_REQUESTED** [review]
This looks OK to me, but I did notice a few minor things that need fixing.

File: .github/workflows/build.yml
Comment (dav3r) [inline .github/workflows/build.yml]: Please make this comment more specific, like you did in https://github.com/cisagov/con-pca-web/pull/145/commits/34662758d1eecabecd92f9086226f3168d2b6dc2.

File: .gitignore
Comment (dav3r) [inline .gitignore]: Please sort this file.  It's fine to keep it separated by sections, but arrange the sections alphabetically, then sort within each section.

File: requirements.txt
Comment (dav3r) [inline requirements.txt:70]: Can these versions be less restrictive, as was done in https://github.com/cisagov/con-pca-api/pull/215/commits/79d38cd3d748f7cfbfa691254e4f84cb575e0f5f?

File: src/version.txt
Comment (dav3r) [inline src/version.txt]: Feels like we are getting closer to version `1.0.0` or at the very least `0.1.0`. 😄 
https://semver.org/

File: tests/container_test.py
Comment (dav3r) [inline tests/container_test.py]: Replace template text from skeleton:
```suggestion
"""Tests for Domain Manager API container."""
```

File: tests/container_test.py
Comment (dav3r) [inline tests/container_test.py]: The text from the skeleton isn't very good here- it should probably be changed to something like this:
```suggestion
    """Verify that the correct number of containers started."""
```

**Review by dav3r — CHANGES_REQUESTED** [review]
Looking good so far, but I noted some items that will look familiar from other recent skeletonizing PRs.

File: requirements.txt
Comment (zenine07) [reply requirements.txt]: @jsf9k Yes, at the moment it could be moved to requirements-test.txt, but we actually had a new request that we'll be utilizing this library in our actual app, so I'm moving it here for now.

Comment (jsf9k) [reply requirements.txt]: OK, got it.

File: src/utils/proxies/websense.py
Comment (zenine07) [reply src/utils/proxies/websense.py]: Fixed in this commit https://github.com/cisagov/domain-manager-api/pull/194/commits/ec801de4b8849f260fd7cdbd3f2ab17c0714e62a

File: src/secrets/quote.txt
Comment (zenine07) [reply src/secrets/quote.txt]: Nope, it's not being used. Removed in https://github.com/cisagov/domain-manager-api/pull/194/commits/11c9b5590641f26779aed76b43f4f0140d93bbe1

File: CONTRIBUTING.md
Comment (zenine07) [reply CONTRIBUTING.md]: Fixed that in https://github.com/cisagov/domain-manager-api/pull/194/commits/bdf4957541d689ea3583c506bc5ecf22ffe5790f

File: .github/workflows/build.yml
Comment (zenine07) [reply .github/workflows/build.yml]: Let me know if this is better https://github.com/cisagov/domain-manager-api/pull/194/commits/8074b12a30685c8a5450cc8c109824df16e86ebf

linux/s390x and linux/ppc64le never actually failed for me, but I started them building over 2 hours ago and never succeeded. As these platforms are older platforms I though it would be fine to disable them to not slow down our development.

File: .gitignore
Comment (zenine07) [reply .gitignore]: Organized and sorted here https://github.com/cisagov/domain-manager-api/pull/194/commits/0ebe5ab8597a61bdb430652da976407ea46e3ccb

File: requirements.txt
Comment (zenine07) [reply requirements.txt:70]: Missed this one while looking over comments. Switched in this commit https://github.com/cisagov/domain-manager-api/pull/194/commits/69c92aa2dad2f2d231599ed26d0930a0cec9e76b

File: src/version.txt
Comment (zenine07) [reply src/version.txt]: Yes, I don't want to go to 1.0.0 until full release, so switched to 0.1.0 for now https://github.com/cisagov/domain-manager-api/pull/194/commits/e1d690f64bc95d6c0ac2c1452652bbef2972080a

File: tests/container_test.py
Comment (zenine07) [reply tests/container_test.py]: Fixed in this commit https://github.com/cisagov/domain-manager-api/pull/194/commits/bdf4957541d689ea3583c506bc5ecf22ffe5790f

File: tests/container_test.py
Comment (zenine07) [reply tests/container_test.py]: Fixed in this commit https://github.com/cisagov/domain-manager-api/pull/194/commits/eff0647ef244b7abef7731b99e59953ca29bf239

File: src/version.txt
Comment (jsf9k) [reply src/version.txt]: Note for the future:  You should use `./bump_version.sh minor`, which will generate a standard commit message for the version bump (and avoid typos, and enforce semantic versioning).

**Comment by jsf9k** [comment]
@zenine07, please re-request my review once all @dav3r's complaints are all put to bed.

File: src/version.txt
Comment (zenine07) [reply src/version.txt]: Awesome. I was not aware of that, thanks.

**Comment by zenine07** [comment]
@dav3r Can you look through the conversations and resolve the ones you think fixed?

File: .github/workflows/build.yml
Comment (dav3r) [reply .github/workflows/build.yml]: That's fine for now- thanks.  I'd love to have a bit more understanding about why those builds are taking so long, but I understand that you can't spend all of your time debugging those other platforms.

**Comment by dav3r** [comment]
> @dav3r Can you look through the conversations and resolve the ones you think fixed?

@zenine07 - I took a look and all of your changes look good.

However, I noticed that in https://github.com/cisagov/domain-manager-api/pull/194/commits/0fec0fabb269e0232a618e65d01b0b73d458c980, you got rid of some of our standard files that still belong in this repo (and every repo that contains Python code):
* `.coveragerc`
* `src/_version.py` (`bump_version.sh` should be updated accordingly and `src/version.txt` can be removed)
* `setup.py`- this file (specifically the `install_requires` line) should contain all of your app-specific requirements that are currently in `requirements.txt`; also, that file has changed lately, so make sure that you grab [the latest version](https://github.com/cisagov/skeleton-python-library/blob/develop/setup.py). 
* `tag.sh`

Please put those files back in and make sure that they are customized (where applicable) for this project.

File: setup.py
Comment (dav3r) [inline setup.py]: @zenine07 - Reminder that you should look for any instances of `example` or `skeleton-python-library` in this file and replace them appropriately.

Comment (zenine07) [reply setup.py]: @dav3r Fixed those references https://github.com/cisagov/domain-manager-api/pull/194/commits/e1fb8c3b4d4c017cc9263e8cb9b68695d99ce056. Let me know if everything is looking good now.

File: setup.py
Comment (dav3r) [inline setup.py:87]: Is there some reason the (non-directory) files in the root of `src/` can't be moved into a `wsgi` directory (or call it whatever makes sense)?  That would more closely align with how the skeleton organizes Python libraries/modules and it would allow you to set up an `entry_point` like [this](https://github.com/cisagov/skeleton-python-library/blob/develop/setup.py#L106-L107).

Related, wherever possible, you should have useful (non-empty, non-stubbed) `__init__.py` and `__main__.py` files.

For a simple (non-skeleton) example to reference, take a look at [`cisagov/hash-http-content`](https://github.com/cisagov/hash-http-content).

Comment (zenine07) [reply setup.py:87]: I can do that. But what is the purpose of the `src/` folder at that point? Wouldn't it be better to just change `src/` to `domain-manager-api/` and put all the files under that? That's the typical structure that you see in all python packages. https://pythonpackaging.info/02-Package-Structure.html

Comment (mcdonnnj) [reply setup.py:87]: The use of the `src/` directory has been a shift in the Python development community that is picking up momentum. The [official packaging guide for Python](https://packaging.python.org/tutorials/packaging-projects/) uses this directory layout now. You can see more discussion points in the [pull request](https://github.com/pypa/packaging.python.org/pull/859) for that change. Although old, [this blog post](https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure) has some excellent points about why the `src/` structure is preferred to a flat structure. Major projects like the [cryptography](https://github.com/pyca/cryptography) package and organizations like NASA are [using this structure](https://github.com/nasa/fprime/tree/devel/Gds).

The long and the short of it is this is now the preferred way to structure Python packages and it has technical merit, so we use that structure.

Comment (zenine07) [reply setup.py:87]: @mcdonnnj That is good to know. I've never knew that Python had an official guide for that. I've never found anything for it and in the pass I've only ever seen not using the `src/` structure. 

So are we also expected to make all of our applications we write packageable? I thought that's the main purpose of include the `setup.py` is for the purposes of packaging and publishing to Python Package Index. I can definitely see the use case for it when I get to skeletonizing the domain-manager-cli project, but in this case, this is an application that is intended to ship as a docker container.

Comment (mcdonnnj) [reply setup.py:87]: Making it a package means you can leverage `pip` to install the project and any of its dependencies as one action. As an example I use `pip` to install [cisagov/hash-http-content](https://github.com/cisagov/hash-http-content) in [cisagov/vdp-scanner-docker](https://github.com/cisagov/vdp-scanner-docker) by including it as a project requirement [here](https://github.com/cisagov/vdp-scanner-docker/blob/e56858082cbd4883afdddf3edb6468b3e6a4b725/src/requirements.txt#L2). You can safely ignore the use of `Pipfile` and [pipenv](https://github.com/pypa/pipenv) in that project as I was experimenting with that as a dependency control mechanism.

We use `pip` to install our projects whether they are deployed in Docker containers or built into AWS AMIs whenever possible as it reduces the complexity of installation and maintenance. Being able to distribute them using pypi is just a side  effect of packaging a Python project.

Comment (zenine07) [reply setup.py:87]: Ok, cool, I'll get onto all that, I've just never heard of a flask application being distributed with `pip` before. So that's the confusion I have with it. I like to know the "why" to a lot of stuff, so hence all my questions. I don't think I have anymore at this point.

Comment (mcdonnnj) [reply setup.py:87]: Although it is an old project that does not conform to our current development practices, we do exactly that with [cisagov/ncats-webd](https://github.com/cisagov/ncats-webd)!

Comment (zenine07) [reply setup.py:87]: Thanks for that code reference. Looking at it and looks like it will be a great help. Thanks for the responses and enlightening me on a few things. Definitely learned some stuff.

Comment (felddy) [reply setup.py:87]: @zenine07 thanks for the great questions, and @mcdonnnj thanks for the thorough answers!  I was tempted to pull "a dad" on Friday and say, "because that's the way it is!". 

Strong work all.  This project is developing a shine!  

💪 ✨

Comment (zenine07) [reply setup.py:87]: @dav3r @mcdonnnj I restructured the project as requested in https://github.com/cisagov/domain-manager-api/pull/194/commits/ba1b4e05b9542313b3d5aefbcf238318ec886839. 

Let me know if this looks good to you now. If so, I'll rerequest reviews.

File: setup.py
Comment (dav3r) [inline setup.py:81]: Why did you remove `coveralls` here?

Comment (zenine07) [reply setup.py:81]: My bad. I'll add it back in. I just put everything from the requirements-test.txt file into that section.

Comment (zenine07) [reply setup.py:81]: Added back in https://github.com/cisagov/domain-manager-api/pull/194/commits/bb0d690ef316e5c0519d27ead1aa253d4021537f

Comment (dav3r) [reply setup.py:81]: Excellent, thanks.

**Review by dav3r — APPROVED** [review]
@zenine07 - I think you have addressed everything that I mentioned and this is all looking great to me now!  🏎️ 

@jsf9k and @mcdonnnj - please take another look here.

File: setup.py
Comment (dav3r) [reply setup.py:87]: Looks great now- thanks! 👍

**Comment by dav3r** [comment]
@zenine07 - One other small thing; please update the PR description:
* Remove "Screenshots" section (it's not needed here)
* In the "Types of Changes " section, add a new checkbox "Merge with skeleton" and check it
* In the "Checklist" section, remove any checkboxes that don't apply and check any remaining ones that do; all remaining checkboxes should be checked at this point

**Comment by zenine07** [comment]
> @zenine07 - One other small thing; please update the PR description:
> 
> * Remove "Screenshots" section (it's not needed here)
> * In the "Types of Changes " section, add a new checkbox "Merge with skeleton" and check it
> * In the "Checklist" section, remove any checkboxes that don't apply and check any remaining ones that do; all remaining checkboxes should be checked at this point

Done.

**Comment by jsf9k** [comment]
I don't think this repo will get upstream updates for some files that are removed in commit 0fec0fa and re-added in 1488060.  That essentially breaks the link with the history from the skeleton.  I'll fix that.

@zenine07 - are there any other places where skeleton files are removed and later re-added that I should look at?

**Comment by zenine07** [comment]
> I don't think this repo will get upstream updates for some files that are removed in commit [0fec0fa](https://github.com/cisagov/domain-manager-api/commit/0fec0fabb269e0232a618e65d01b0b73d458c980) and re-added in [1488060](https://github.com/cisagov/domain-manager-api/commit/1488060e7314fe387b900cc23c7bfe1f9f580053). That essentially breaks the link with the history from the skeleton. I'll fix that.
> 
> @zenine07 - are there any other places where skeleton files are removed and later re-added that I should look at?

@jsf9k Just did a quick lookover on the commits and it doesn't look like it. But, I actually think those commits you're referencing are not skeleton files. Those files do no exist in skeleton-docker, so wouldn't break the link with the history from the skeleton.

**Comment by jsf9k** [comment]
> > I don't think this repo will get upstream updates for some files that are removed in commit [0fec0fa](https://github.com/cisagov/domain-manager-api/commit/0fec0fabb269e0232a618e65d01b0b73d458c980) and re-added in [1488060](https://github.com/cisagov/domain-manager-api/commit/1488060e7314fe387b900cc23c7bfe1f9f580053). That essentially breaks the link with the history from the skeleton. I'll fix that.
> > @zenine07 - are there any other places where skeleton files are removed and later re-added that I should look at?
> 
> @jsf9k Just did a quick lookover on the commits and it doesn't look like it. But, I actually think those commits you're referencing are not skeleton files. Those files do no exist in skeleton-docker, so wouldn't break the link with the history from the skeleton.

Ah, OK.  You are correct.

**Review by jsf9k — APPROVED** [review]

**Comment by jsf9k** [comment]
Need a review from @inlguy before we can merge this one.

**Review by inlguy — APPROVED** [review]

**Pull request merged**
